lib/LANraragi/Controller/Api/Tankoubon.pm

```diff
@@ -0,0 +1,122 @@
+package LANraragi::Controller::Api::Tankoubon;
+use Mojo::Base 'Mojolicious::Controller';
+
+use Redis;
+use Encode;
+
+use LANraragi::Model::Tankoubon;
+use LANraragi::Utils::Generic qw(render_api_response);
+
+sub get_tankoubon_list {
+
+    my $self = shift;
+    my $req  = $self->req;
+
+    my $page       = $req->param('page');
+    $page //= 0;
+
+    my ( $total, $filtered, @rgs ) = LANraragi::Model::Tankoubon::get_tankoubon_list($page);
+    $self->render( json => {result => \@rgs, total => $total, filtered => $filtered} );
+
+}
+
+sub get_tankoubon {
+
+    my $self     = shift;
+    my $tank_id    = $self->stash('id');
+    my $req  = $self->req;
+
+    my $decoded       = $req->param('decoded');
+    $decoded //= 0;
```

Comment:
Are those default values required here considering the sub signature for the Model also has defaults?  

Ditto for get_list above.

Reply:
I removed them, but now the console throws a lot of warnings with `Use of uninitialized value $page in multiplication (*) at /home/koyomi/lanraragi/script/../lib/LANraragi/Model/Tankoubon.pm`, doesn't cause any error, but it's probably annoying for who is trying to debug. Not sure how to fix that.

Reply:
Looked at the docs for sub signatures again, the default values are only applied if you don't give it a parameter, but in this case we're going to give them `undef`..  

But that can be fixed by using `//=` in the signature: 
> A default value expression can also be written using the //= operator, where it will be evaluated and used if the caller omitted a value or the value provided was undef.

```
sub foo ($name //= "world") {
    print "Hello, $name";
}

foo(undef);  # will print "Hello, world"
```

Reply:
Tried this, but trying to use //= in the signature throwed errors... I'm guessing that what you said should work, but maybe hasn't been implemented correctly(Or maybe I didn't), anyways, just to supress that warning for the moment I added a `$page //= 0 ` inside those subs, which is basically what I did at the start, but I guess it works until signatures gets out of experimental phase.

Reply:
Ah, maybe that depends on perl versions, yeah you did the right thing by adding the default assignation back!